src/App.tsx

```diff
@@ -15,48 +16,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goodsFromServer];
+
+  visibleGoods.sort((a, b) => {
+    if (sortType === SortType.ALPHABET) {
+      return a.localeCompare(b);
+    }
+
+    if (sortType === SortType.LENGTH) {
+      return a.length - b.length;
+    }
+
+    return 0;
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, handleSort] = useState(SortType.NONE);
```

Comment:
Typically we use a convention to name `useState` variables as: `[sortType, setSortType]` see [here](https://mate-academy.github.io/fe-program/js/extra/react-typescript#usestate) 
we use a prefix `handle` for custom methods.